Frontend/student-hub/src/pages/CoursePage.js

```diff
@@ -36,7 +36,30 @@ function CoursePage() {
     const [currentComments, setCurrentComments] = useState([]);
     const [newComment, setNewComment] = useState('');
     const [currentPostId, setCurrentPostId] = useState(null);
+    const [media, setMedia] = useState(null);
+    const [selectedEmoji, setSelectedEmoji] = useState({});
+    const [emojis, setEmojis] = useState([])
 
+// emoji
+    useEffect(() => {
+        const fetchEmojis = async () => {
+            try{
+
+                const response = await fetch(`${process.env.REACT_APP_API}/emoji`);
+
+                const response = await fetch('http://localhost:8080/emoji');
+                if (!response.ok) {
+                    throw new Error('Failed to fetch emojis');
+                }
+                const emojistData = await response.json();
+                setEmojis(emojistData);
+            } catch (error) {
+                console.error('Error fetching emojis:', error);
+            }
```

Comment:
don't love the try catch happening within the same code block. If you know something is wrong you can just do the appropriate action there